docs/concepts/overview/extending.md

```diff
@@ -0,0 +1,217 @@
+
+{% endcapture %}
+
+
+{% capture body %}
+
+## Overview
+
+Customization approaches can be broadly divided into **configuration**, which only involves changing flags, local configuration files, or API resources; and **extensions**, which involve running additional programs or services.  This document is primarily about extensions.
+
+## Configuration
+
+**Configuration files** and **flags** are documented in the Reference section of the online documentation, under each binary:
+
+* [kubelet](/docs/admin/kubelet/)
+* [kube-apiserver](/docs/admin/kube-apiserver/)
+* [kube-controller-manager](/docs/admin/kube-controller-manager/)
+* [kube-scheduler](/docs/admin/kube-scheduler/).
+
+Flags and configuration files may not always be changable in a hosted Kubernetes service or a distribution with managed installation.  When they are changable, they are usually only changeable by the cluster administrator.    Also, they are subject to change in future Kubernetes versions, and setting them may require restarting processes.  For those reasons, they should be used only when there are no other options.
+
+**Built-in Policy APIs**, such as [ResourceQuota](/docs/concepts/policy/resource-quotas/), [PodSecurityPolicies](/docs/concepts/policy/pod-security-policy/), [NetworkPolicy](/docs/concepts/services-networking/network-policies/) and Role-based Access Control ([RBAC](/docs/admin/authorization/rbac/)), are built-in Kubernetes APIs.  APIs are typically used with hosted Kubernetes services, and with managed Kubernetes installations.   They are declarative and use the same conventions are other Kubernetes resources like pods, so new cluster configuration can be repeatable and be managed the same way as applications.  And, where they are stable, they enjoy a [defined support policy](/docs/reference/deprecation-policy/) like other Kubernetes APIs.  For these reasons, they are preferred over **configuration files** and **flags** where suitable.
+
+The remainder of this guide is about Extensions.
+
+## Extensions
+
+Extensions are software components that extend and deeply integrate with kubernetes. 
+They adapt it to support new types and new kinds of hardware.
+
+Most cluster administrators will use a hosted or distribution 
+instance of Kubernetes.  As a result, most Kubernetes users will need to 
+install extensions and fewer will need to author new ones.
+
+## Extension Patterns
+
+Kubernetes is designed to be automated by writing client programs.  Any
+program that reads and/or writes to the Kubernetes API can provide useful
+automation.  **Automation** can run on the cluster or off it.  By following
+the guidance in this doc you can write highly available and robust automation.
+Automation generally works with any Kubernetes cluster, including hosted
+clusters, and managed installations.
+
+There is a specific pattern for writing client programs that work well with
+Kubernetes called the **Controller** pattern.  Controllers typically read an
```

Comment:
"Controller" should be in italics instead of in bold. When introducing a new term, the convention is to use italics instead of bold.